### ISSUE TITLE
Aws Implementation Issue with Custom domain with path

### ISSUE THREAD

**Issue by biswajitdas** [simov/grant#192]
I am trying to use same custom domain name for stage and production separated by path but Grant is failing in that case. example

```
https://custom.domain/stage --> Stage Lambda (Stage : $default)
https://custom.domain/prod --> Prod Lambda (Stage : $default)
```

Settings used for prod
```
{
    "defaults": {
      "origin": "https://custom.domain",
      "prefix": "/prod",
      "transport": "session",
      "state":true
    },
    "google": {
      "authorize_url": "https://accounts.google.com/o/oauth2/auth",
      "access_url": "https://accounts.google.com/o/oauth2/token",
      "oauth": 2,
      "scope_delimiter": " ",
      "pkce": true,
      "scope": [
        "profile",
        "email",
        "openid"
      ],
      "response": ["tokens", "profile"],
      "nonce": true,
      "custom_params": {
        "access_type": "offline",
        "prompt": "consent"
      },
      "callback": "https://another.domain"
    }
  }
```

Settings used for prod
```
{
    "defaults": {
      "origin": "https://custom.domain",
      "prefix": "/stage",
      "transport": "session",
      "state":true
    },
    "google": {
      "authorize_url": "https://accounts.google.com/o/oauth2/auth",
      "access_url": "https://accounts.google.com/o/oauth2/token",
      "oauth": 2,
      "scope_delimiter": " ",
      "pkce": true,
      "scope": [
        "profile",
        "email",
        "openid"
      ],
      "response": ["tokens", "profile"],
      "nonce": true,
      "custom_params": {
        "access_type": "offline",
        "prompt": "consent"
      },
      "callback": "https://another.domain"
    }
  }
```
Event version used : 1.0

What did I see?
```
path = '/prod/google'
requestContext.path = '/google'
```

I saw in grant library it always expects that to be `'/prod/google'` for `requestContext.path`. As of now I am overriding that value inside from my script as `event.requestContext.path = event.path;`.
Can you please suggest me for better solution?

Thanks.

**Comment by simov** [comment]
Seems like another edge case for the input event data structure. I have to re-create that setup on my end in order to test it properly.

**Comment by simov** [comment]
Just released v5.4.1 where all of this should be fixed.

Here are the tests that I did:

Default domain with Grant prefix `/dev/connect`:

```
https://[ID].execute-api.[REGION].amazonaws.com/dev/connect/google
```

Custom domain without custom path mapping with Grant prefix `/connect`:

```
https://custom-domain.com/connect/google
```

Custom domain with custom path mapping with Grant prefix `/v1/connect`:

```
https://custom-domain.com/v1/connect/google
```

Note that I also added support for Event v2, so you can use that as well. However, I found a bug when Event v2 is used with Custom domain + custom path - the custom path is never sent to your lambda, and so only in that case the Grant prefix have to be `/connect` for:

```
https://custom-domain.com/v1/connect/google
```

And you have to set the `redirect_uri` explicitly as `https://custom-domain.com/v1/connect/google/callback`.

The reason why is because AWS never sends the custom prefix in Event v2, even when using the `rawPath`, which is clearly a bug.

Let me know if it works for you.

**Comment by simov** [comment]
Pushed another fix in v5.4.2 with a bit more streamlined [path matching logic](https://github.com/simov/grant-aws#routes).
